Tooltips / Popper support

micolous opened the pull request:
Based on ~~#45~~ ~~#48~~ #52

This adds support for tooltips with [`popper-rs`](https://github.com/ctron/popper-rs/).

Bootstrap tooltips are built with Popper. Both Yew and Popper assume they have complete control of the DOM, so Popper needs some extra wiring to handle the component lifecycle correctly, which `popper-rs` provides.

This takes some ideas from [`react-bootstrap`'s Tooltip implementation](https://github.com/react-bootstrap/react-bootstrap/blob/master/src/Tooltip.tsx), but goes for a slightly simpler model, where Tooltips are dropped into a portal directly (like [`popper-rs`' example](https://github.com/ctron/popper-rs/blob/main/examples/yew/src/example/basic.rs)), and the Tooltip component is responsible for all the event handling (rather than [`Overlay`](https://github.com/react-bootstrap/react-bootstrap/blob/master/src/Overlay.tsx) and [`OverlayTrigger`](https://github.com/react-bootstrap/react-bootstrap/blob/master/src/OverlayTrigger.tsx)).

To position the tooltip correctly and mark the target element with accessibility attributes, you need to expose a `NodeRef` for each component you want to add a tooltip to. I've exposed `NodeRef`s for `Button`, `FormControl` and `Link`.

One of the complexities with this is deciding when to trigger the tooltip, which is a problem both `react-bootstrap` and Bootstrap never really solved. By default, this will trigger on focus and on pointer hover. I've added options for selectively disabling triggering on focus, but [deficiencies in the Media Queries spec](https://github.com/w3c/csswg-drafts/issues/5462) and [browser implementations](https://github.com/mdn/browser-compat-data/issues/24451) make none of them great options. There are a lot of caveats listed in the docs.

Unlike regular Bootstrap, this doesn't support tooltips from `data` attributes.

I've added examples using all of those controls, and also a direct `ref=` attribute on a HTML elements.

isosphere commented on examples/forms/src/main.rs:
```diff
@@ -372,6 +385,91 @@ impl Component for Model {
+                        {
+                            for [
+                                TooltipFocusTrigger::IfNoHover,
+                                TooltipFocusTrigger::IfNoAnyHover,
```
These variants don't exist!

micolous replied:
Fixed

isosphere reviewed (changes requested):
Please correct the use of the `TooltipFocusTrigger` variants in the forms example. Looks like you may have changed the names and missed updating them.

micolous commented:
Build failure in CI looks like it's running on a newer version of Rust which doesn't work with the old version of `wasm-bindgen`. Will have a look...

micolous commented:
Rebased on top of #52 to update old dependencies in `Cargo.lock`, that seems to fix the build.

I've also added a note about iPad hovering, because the Apple Pencil (stylus) is a bit surprising.

isosphere commented:
The examples work great! Thanks for the valuable contribution.
